### PR TITLE
Crash when deleting buffers

### PR DESCRIPTION
**Pull request by kkaefer**
I'm seeing a fairly frequent crash bug that happens randomly during buffer deletion. The buffer IDs are valid and do exist.

What could be happening is that we have an OpenGL context loss that we're ignoring, which invalidates all of the buffers.

buffer hpp 2015-03-17 18-10-54

### CODE REVIEW

**Comment by kkaefer** [comment]
~LiveTileData is run in the map thread

**Comment by incanus** [comment]
Cross-ref to https://github.com/mapbox/mapbox-gl-native/issues/1022 — I will double-check that things are kosher here.

**Comment by picciano** [comment]
+1 Awesome, no more crashing. Thanks!

File: src/mbgl/map/map.cpp [hunk 1]
```diff
@@ -113,7 +127,6 @@ void Map::start(bool startPaused) {
 
         // Remove all of these to make sure they are destructed in the correct thread.
         style.reset();
-        workers.reset();
```
Comment (kkaefer) [inline src/mbgl/map/map.cpp]: This crashes in Debug mode since the Worker threads are now joined from the wrong thread.

Comment (kkaefer) [inline src/mbgl/map/map.cpp]: Fixed in 9d30d4042c3311d02568adbb38dd13fb82faf75d

**Pull request merged**